vms/platformvm/state.go

```diff
@@ -389,25 +389,28 @@ func (vm *VM) removeUTXO(db database.Database, utxoID ids.ID) error {
 // Only returns UTXOs after [start].
 // Returns at most [limit] UTXO IDs.
 // Returns nil if no UTXOs reference [addr].
-func (vm *VM) getReferencingUTXOs(db database.Database, addr []byte, start ids.ID, limit int) (ids.Set, error) {
-	toFetch := limit
-	utxoIDs := ids.Set{}
+func (vm *VM) getReferencingUTXOs(db database.Database, addr []byte, start ids.ID, limit int) ([]ids.ID, error) {
+
+	idSlice := []ids.ID(nil)
+
 	iter := prefixdb.NewNested(addr, db).NewIteratorWithStart(start.Bytes())
 	defer iter.Release()
-	for toFetch > 0 && iter.Next() {
-		if utxoID, err := ids.ToID(iter.Key()); err != nil {
+	numFetched := 0
+	for numFetched < limit && iter.Next() {
+		if keyID, err := ids.ToID(iter.Key()); err != nil {
 			return nil, err
-		} else if !utxoID.Equals(start) {
-			utxoIDs.Add(utxoID)
-			toFetch--
+		} else if !bytes.Equal(keyID.Bytes(), start.Bytes()) {
+			idSlice = append(idSlice, keyID)
+			numFetched++
 		}
 	}
-	return utxoIDs, nil
+	return idSlice, nil
 }
 
 // Persist that the UTXO with ID [utxoID] references [addr]
 func (vm *VM) putReferencingUTXO(db database.Database, addrBytes []byte, utxoID ids.ID) error {
 	prefixedDB := prefixdb.NewNested(addrBytes, db)
+	defer prefixedDB.Close()
```

Comment:
Since we are closing the DB here correctly now, we should probably handle the error from the Close call as well.